File: app/src/main/java/com/example/databinding/ViewModels/MainActivityViewModel.java
package com.example.databinding.ViewModels;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;
import com.example.databinding.Models.UserModel;

public class MainActivityViewModel extends ViewModel {

    public MutableLiveData<UserModel> userModelMutableLiveData;

    public MainActivityViewModel(){
        userModelMutableLiveData = new MutableLiveData<>();
    }

    public void loginUser(String userName, String password){
        if(userName != null && password != null){
            UserModel userModel = new UserModel();
            userModel.setUserName(userName);
            userModel.setPassword(password);
            userModelMutableLiveData.setValue(userModel);
        }

    }
}

File: app/src/main/java/com/example/databinding/MainActivityViewModel.kt
package com.example.databinding

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.example.databinding.Models.UserModel

class MainActivityViewModel : ViewModel() {
    var userModelMutableLiveData: MutableLiveData<UserModel>
    var userName = MutableLiveData<String>()
    var password = MutableLiveData<String>()
    fun loginUser() {
        val userModel = UserModel(userName.value!!, password.value!!)
        userModelMutableLiveData.value = userModel
    }

    init {
        userModelMutableLiveData = MutableLiveData()
    }
}